### PR TITLE
Fixes to Hallow's End

### PR DESCRIPTION
**Pull request by GuybrushGit**
https://github.com/LightsHope/server/issues/53  https://github.com/LightsHope/server/issues/42

- Added missing NPC Katrina Shimmerstar
- Some innkeepers were missing the "Trick or Treat" gossip
- Fixed Bobbing Apple giving Well Fed
- Switched Traveler's Backpacks with Pumpkin Bags when event is active.

### CODE REVIEW

**Comment by Cabro** [comment]
Fix the treat bags being mostly empty :(

**Comment by brotalnia** [comment]
You used the wrong spell id for the well fed buff. The correct one is 24870 which gives 25% to stamina and spirit.

https://web.archive.org/web/20060614014216/http://wow.allakhazam.com:80/db/item.html?witem=20516

Posted @ Thu, Oct 20th 5:18 PM 2005
> It doesn't say so on the description, but the Well Fed Bonus is a 25% bonus to Stamina and Spirit for 15 minutes.

**Comment by GuybrushGit** [comment]
@brotalnia changed it, but that spell is not doing what it says it does, it adds 6 spirit and 10% of level stamina.

@Cabro thats not a bug, its just because the event was accidentally started on elysium where some of the items dont exist yet

**Comment by brotalnia** [comment]
I guess blizzard tricked us. It could be intentional.

**Comment by thepparker** [comment]
Which items are missing from the treat bags? If they are event specific, then the patch level should be removed from the item.

**Comment by Cabro** [comment]
@GuybrushGit but i play on Anathema. And out of 5 bags, 4 had absolutely nothing in them. It should have at least 1 item in them, like lockboxes?

**Comment by GuybrushGit** [comment]
Adjusted the loot chances and set it to group 1 to make it always drop loot.

File: sql/migrations/20171020210201_world.sql [hunk 1]
```diff
@@ -0,0 +1,818 @@
...
+REPLACE INTO `gossip_menu_option` VALUES (1290, 0, 0, 'Trick or Treat!', 1, 1, -1, 0, 342, 0, 0, NULL, 164);
+REPLACE INTO `gossip_menu_option` VALUES (1294, 0, 0, 'Trick or Treat!', 1, 1, -1, 0, 342, 0, 0, NULL, 164);
+REPLACE INTO `gossip_menu_option` VALUES (6525, 0, 0, 'Trick or Treat!', 1, 1, -1, 0, 342, 0, 0, NULL, 164);
+UPDATE `creature_template` SET `gossip_menu_id`=342 WHERE  `entry`=16256 AND `patch`=0;
+UPDATE `gossip_menu_option` SET `action_menu_id`=-1, `action_script_id`=342, `condition_id`=164 WHERE  `menu_id`=9868 AND `id`=0;
+
+-- Treat Bag loot chances
+UPDATE `item_loot_template` SET `groupid`=1 WHERE `entry`=20393;
+UPDATE `item_loot_template` SET `ChanceOrQuestChance`=0 WHERE `entry`=20393;
+UPDATE `item_loot_template` SET `ChanceOrQuestChance`=15 WHERE `entry`=20393 AND `item` IN (20390,20388,20389);
```
Comment (thepparker) [inline sql/migrations/20171020210201_world.sql]: 20413 should be around 15% too

Comment (GuybrushGit) [reply sql/migrations/20171020210201_world.sql]: Adjusted it according to this: https://web.archive.org/web/20081017201418/http://thottbot.com:80/i20393

The candy is ungrouped at 66%, random wand in group 1 at 15% and the rest in group 1 at 2%.

**Pull request merged**